### PR TITLE
Add support constructor parameters annotations for schemes

### PR DESCRIPTION
**Pull request by altro3**

### CODE REVIEW

**Comment by sdelamo** [comment]
@altro3 you have to sign the CLA again. [Sorry for the inconvenience](https://micronaut.io/2024/02/06/micronaut-contributor-license-agreement-cla-update/).

**Comment by altro3** [comment]
@sdelamo ??? What does it mean? Is this blocking the merge? What do I need to do?

**Comment by sdelamo** [comment]
> @sdelamo ??? What does it mean? Is this blocking the merge? What do I need to do?

No action in your side. It appears to be a glitch with cla-assistant.io reporting. Sorry for the noise.

**Comment by sdelamo** [comment]
@altro3 Can you merge master into this branch?

**Comment by altro3** [comment]
@sdelamo done

File: openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautKotlinCodegen.java [hunk 1]
```diff
@@ -902,6 +902,27 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
...
         op.vendorExtensions.put("originReturnProperty", op.returnProperty);
+        if (op.responses != null && !op.responses.isEmpty()) {
+            for (var resp : op.responses) {
+                if (resp.isDefault) {
+                    resp.code = "default";
+                }
+            }
+        }
+        /*
+        0 = {LoadedVisitor@8165} "io.micronaut.openapi.visitor.OpenApiGroupInfoVisitor@3d615b2e"
```
Comment (dstepanov) [inline openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautKotlinCodegen.java:913]: Pls remove

File: openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java [hunk 2]
```diff
@@ -219,12 +221,16 @@ public static boolean isIgnoredParameter(TypedElement parameter) {
...
-            || isAnnotationPresent(parameter, "org.springframework.web.bind.annotation.SessionAttribute")
-            || isAnnotationPresent(parameter, "org.springframework.web.bind.annotation.SessionAttributes")
+            || isParamAnnotationPresent(parameter, "io.micronaut.session.annotation.SessionValue")
+            || isParamAnnotationPresent(parameter, "org.springframework.web.bind.annotation.SessionAttribute")
+            || isParamAnnotationPresent(parameter, "org.springframework.web.bind.annotation.SessionAttributes")
             || isIgnoredParameterType(parameter.getType());
     }
 
+    private static boolean isParamAnnotationPresent(Element element, String className) {
+        return element.findAnnotation(className).isPresent();
```
Comment (dstepanov) [inline openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: There is not hasAnnotation

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java (hunk 2 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: @dstepanov ??? What do you mean? How I need to change it?

Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: do I need to change it  to this:
```java
parameter.isAnnotationPresent("org.springframework.web.bind.annotation.SessionAttribute")
```
?

Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: and what difference between `hasAnnotation` and `isAnnotationPresent` ?

Comment (dstepanov) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: I think it’s better to avoid returning an optional

**Review by dstepanov — APPROVED** [review]

File: openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java (hunk 2 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java:231]: I'm sorry, I still didn't get it.: Do I need to change something? If so, what should I change?

**Comment by altro3** [comment]
@sdelamo could you merge this last PR and finalize release?

File: openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java (hunk 2 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ElementUtils.java]: @dstepanov ok, I changed it to `hasAnnotation`. But what difference between `hasAnnotation` and `isAnnotationPresent` ?

**Comment by sdelamo** [comment]
@altro3 can you merge master into this branch?

**Comment by altro3** [comment]
@sdelamo done

**Pull request merged**